pkg/library/reconcilers/fetcher.go

```diff
@@ -41,6 +44,14 @@ func fetchGateway(ctx context.Context, k8sClient client.Reader, key client.Objec
 		return nil, err
 	}
 
+	return gw, nil
+}
+
+func fetchProgrammedGateway(ctx context.Context, k8sClient client.Reader, key client.ObjectKey) (*gatewayapiv1.Gateway, error) {
+	gw, err := fetchGateway(ctx, k8sClient, key)
+	if err != nil {
+		return nil, err
+	}
 	if meta.IsStatusConditionFalse(gw.Status.Conditions, string(gatewayapiv1.GatewayConditionProgrammed)) {
 		return nil, fmt.Errorf("gateway (%v) not ready", key)
 	}
```

Comment:
https://github.com/Kuadrant/kuadrant-operator/pull/638#discussion_r1642706289 suggests aligning this with the same check in the DAG
```suggestion
	if !meta.IsStatusConditionTrue(gw.Status.Conditions, string(gatewayapiv1.GatewayConditionProgrammed)) {
		return nil, fmt.Errorf("gateway (%v) not ready", key)
	}
```
But this change would cause the DNSPolicy tests to fail, so either:
- DNSPolicy should also not target only programmed gateways
or
- Update the DNSPolicy tests so that the created Gateway has the `Programmed: true` status

Is this something we want to do and whats the best path to reflect the flow?

Reply:
I was originally plannning to do this as part of this PR, but probably best as to seperate it to it's own now. Yeah, I think all these checks should use `!meta.IsStatusConditionTrue` instead since `meta.IsStatusConditionFalse` will be `false` for if state is unknown or if the gateway does not have the programmed condition at all. Meaning, DNSPolicy/AuthPolicy/RateLimitPolicy can reconcile on these resources which is probably not intended.